packages/common/src/functions.ts

```diff
@@ -90,3 +90,24 @@ export function enforceSequentialExecution<P, T extends (...args: any[]) => Prom
         return queue;
     }) as T;
 }
+
+/**
+ * 
+ * @param fn a function to memoize
+ * @param argsHash a function that returns a string hash for the arguments, defaults to JSON.stringify
+ * @returns a memoized version of `fn`
+ */
+export function memoize<T extends (...args: any[]) => any>(
```

Comment:
when using this generic approach I think we must open a strategy to clear the cache.
or use a WeakMap but it should have different way to do args check which is more complex.

for now I would expose the internal cache on the memoize function itself and I would go and maybe add an LRU example of usage. otherwise it is dangers to use.

Reply:
also we can maybe make a better type here that will represent the argsHash function better.

Reply:
where do you want to use it. I know we have few running around. and I wrote one specific for strings arguments with no cache handling but it's for tests